week_4/practice/todo/consumer/consumer_test.go

```diff
@@ -0,0 +1,199 @@
+			return nil
+		}
+
+		if err := pact.Verify(test); err != nil {
+			log.Fatalf("Error on Verify: %v", err)
+		}
+	})
+
+	t.Run("TestUpdateTodo", func(t *testing.T) {
+		id := "id1"
+		pact.AddInteraction().Given("Update todo with id = id1").UponReceiving("A request to list todo").
+			WithRequest(dsl.Request{
+				Method: http.MethodPut,
+				Path:   dsl.String(fmt.Sprintf("/v1/todo/%s", id)),
+			}).WillRespondWith(dsl.Response{
+			Status:  http.StatusOK,
+			Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+			Body: dsl.Like(map[string]interface{}{
+				"responseStatus": dsl.Like(map[string]interface{}{
+					"code": dsl.Like(int32(http.StatusOK)), "message": dsl.Like(""),
+				}),
+			}),
+		})
+
+		test := func() (err error) {
+			proxy := ToDoProxy{Host: "localhost", Port: pact.Server.Port}
+			resp, err := proxy.UpdateToDo(id, UpdateTodoRequest{Title: "only update title", Completed: true, Description: ""})
+			if err != nil {
+				return err
+			}
+			fmt.Println(resp)
+			return nil
+		}
+
+		if err := pact.Verify(test); err != nil {
+			log.Printf("Error on Verify: %v", err)
+		}
+	})
+
+	t.Run("TestDeleteTodo", func(t *testing.T) {
+		id := "id1"
+		pact.AddInteraction().Given("Delete todo with id = id1").UponReceiving("A request to list todo").
+			WithRequest(dsl.Request{
+				Method: http.MethodDelete,
+				Path:   dsl.String(fmt.Sprintf("/v1/todo/%s", id)),
+			}).WillRespondWith(dsl.Response{
+			Status:  http.StatusOK,
+			Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+			Body: dsl.Like(map[string]interface{}{
+				//test another way to create data response body
+				"responseStatus":map[string]interface{}{
+					"name":int32(http.StatusOK) ,
+					"message": "",
+				},
+			}),
+		})
+
+		test := func() (err error) {
+			proxy := ToDoProxy{Host: "localhost", Port: pact.Server.Port}
+			resp, err := proxy.DeleteToDo(id)
+			if err != nil {
+				return err
+			}
+			fmt.Println(resp)
+			return nil
+		}
+
+		if err := pact.Verify(test); err != nil {
+			log.Printf("Error on Verify: %v", err)
+		}
+	})
+
+	t.Run("TestGetTodo", func(t *testing.T) {
+		id := "id1"
+		pact.AddInteraction().Given("Get todo with id = id1").UponReceiving("A request to list todo").
+			WithRequest(dsl.Request{
+				Method: http.MethodGet,
+				Path:   dsl.String(fmt.Sprintf("/v1/todo/%s", id)),
+			}).WillRespondWith(dsl.Response{
+			Status:  http.StatusOK,
+			Headers: dsl.MapMatcher{"Content-Type": dsl.String("application/json")},
+			Body: dsl.Like(map[string]interface{}{
+				"todo": dsl.Like(map[string]interface{}{"id": dsl.Like("id1"), "title": dsl.Like("ToDo A")}),
+				"responseStatus": map[string]interface{}{
+					"name":    int32(http.StatusOK),
+					"message": "",
+				},
+			}),
+		})
+
+		test := func() (err error) {
+			proxy := ToDoProxy{Host: "localhost", Port: pact.Server.Port}
+			resp, err := proxy.GetToDo(id)
+			if err != nil {
+				return err
+			}
+			fmt.Println(resp)
```

Comment:
Instead of logging resp here, we could assert the response against the expected value.